src/main/java/com/denizenscript/ddiscordbot/commands/DiscordInteractionCommand.java

```diff
@@ -0,0 +1,564 @@
+
+import org.bukkit.Bukkit;
+
+import net.dv8tion.jda.api.JDA;
+import net.dv8tion.jda.api.entities.Message;
+import net.dv8tion.jda.api.entities.MessageEmbed;
+import net.dv8tion.jda.api.interactions.Interaction;
+import net.dv8tion.jda.api.interactions.InteractionHook;
+import net.dv8tion.jda.api.interactions.commands.Command;
+import net.dv8tion.jda.api.interactions.commands.OptionType;
+import net.dv8tion.jda.api.interactions.commands.build.CommandData;
+import net.dv8tion.jda.api.interactions.commands.build.OptionData;
+import net.dv8tion.jda.api.interactions.commands.build.SubcommandData;
+import net.dv8tion.jda.api.interactions.commands.privileges.CommandPrivilege;
+import net.dv8tion.jda.api.interactions.components.ActionRow;
+import net.dv8tion.jda.api.requests.restaction.CommandCreateAction;
+import net.dv8tion.jda.api.requests.restaction.WebhookMessageAction;
+import net.dv8tion.jda.api.requests.restaction.interactions.ReplyAction;
+
+public class DiscordInteractionCommand extends AbstractCommand implements Holdable {
+
+    public DiscordInteractionCommand() {
+        setName("discordinteraction");
+        setSyntax("discordinteraction [id:<id>] [command [create/perms/delete]/defer/reply/delete] (group:<group>) (name:<name>) (description:<description>) (options:<options>) (enabled:{true}/false) (enable_for:<list>) (disable_for:<list>) (interaction:<interaction>) (ephermal:true/{false}) (attach_file_name:<name>) (attach_file_text:<text>) (rows:<rows>) (<message>)");
+        setRequiredArguments(3, 10);
+    }
+
+    // <--[command]
+    // @Name discordinteraction
+    // @Syntax discordinteraction [id:<id>] [command [create/perms/delete]/defer/reply/delete] (group:<group>) (name:<name>) (description:<description>) (options:<options>) (enabled:{true}/false) (enable_for:<list>) (disable_for:<list>) (interaction:<interaction>) (ephermal:true/{false}) (attach_file_name:<name>) (attach_file_text:<text>) (rows:<rows>) (<message>)
+    // @Required 3
+    // @Maximum 10
+    // @Short Manages Discord interactions.
+    // @Plugin dDiscordBot
+    // @Group external
+    //
+    // @Description
+    // Manages Discord interactions.
+    //
+    // You can manage slash commands using the "command" instruction, specifying further instructions.
+    //
+    // Using the "command" instruction, you can create a new slash command, or edit the permissions of or delete an existing command.
+    //
+    // To create (or delete) a command in a specific Discord guild, use the "group" argument. If not present, a global command will be created. NOTE: Global slash commands take up to an hour to register.
+    // When creating, both a name and description are required.
+    //
+    // The "options" argument controls the command parameters. It is a ListTag of MapTags that can sometimes hold ListTags. It is recommended to use external YAML files as a simpler form of data representation when creating commands. See <@link command yaml>.
```

Comment:
Recommending an external yaml file is a bad idea without very good justification

Reply:
It's very useful for command arguments. I noted this in the 'design challenges' section.

```yml
my_command_options:
- type: string
  name: animal
  description: The animal
  required: true
  choices:
  - name: Dog
    value: animal_dog
  - name: Cat
    value: animal_cat
```

I attempted to recreate this as a tag in one of the examples. It didn't work.

Reply:
Refer to how `attribute_modifiers` in Denizen proper deals with this input mess